test/typeinference/Tests/InferProgramSpec.hs

```diff
@@ -77,3 +87,112 @@ spec = do
         typedFully = lowerAndInfer fully
         typedMinimal = lowerAndInfer minimal
     typedFully `shouldBe` typedMinimal
+
+  {- A list of tricky functions to verify the correctness of type inference.
+
+  For now, I hardcode all the expected results. We should fix that later.
+  Uncomment the print statement to output the inference result for each function.
+  -}
+  it "infers id function" $ do
+    let minimal = parseProgram $ unlines
+          [ "id(x) ="
+          , "  x"
+          ]
+        typedMinimal = lowerAndInfer minimal
+        typedPString = show typedMinimal
```

Comment:
We shouldn't be comparing the string representation of the inferred type when we can just compare the type representation directly.

Reply:
oh wait yikes I see why you're doing this. This seems kind of fragile.

Why not compare an unannotated and fully annotated version? e.g., compare `addOne` to:

```
addOne x : Int -> Int = ((x: Int) + 1):Int
```

I realize that there isn't yet syntax for type variables, so I'll get working on that.